src/Extension/CoreExtension.php

```diff
@@ -686,12 +686,12 @@ public static function merge(...$arrays): array
     /**
      * Slices a variable.
      *
-     * @param mixed $item         A variable
-     * @param int   $start        Start of the slice
-     * @param int   $length       Size of the slice
-     * @param bool  $preserveKeys Whether to preserve key or not (when the input is an array)
+     * @param \Traversable|array|string $item         A variable
+     * @param int                       $start        Start of the slice
+     * @param int                       $length       Size of the slice
+     * @param bool                      $preserveKeys Whether to preserve key or not (when the input is an array)
      *
-     * @return mixed The sliced variable
+     * @return ($item is string ? string : array)
```

Comment:
This could be more precise for the case of array as well, by making it generic:

```
/**
 * @template T
 * @param \Traversable<T>|T[]|string $item
 * @return ($item is string ? string : T[])
 */
```

This would allow TwigStan to preserve the type of item values.

Reply:
Good idea. I updated the PR so that it also handles TKey.

Reply:
The handling of TKey is wrong. It will produce bad results when phpstan infers the key type as a union of literals rather than as `int`. phpstan uses a custom type extension to implement the return type of `array_slice`, not phpdoc.

Reply:
Ok, I changed it back to value only. We can address this later with an extension in TwigStan to further refine it like PHPStan does.